runtime/lua/vim/lsp/util.lua

```diff
@@ -806,14 +806,22 @@ function M.convert_input_to_markdown_lines(input, contents)
     assert(type(input) == 'table', "Expected a table for Hover.contents")
     -- MarkupContent
     if input.kind then
-      -- The kind can be either plaintext or markdown. However, either way we
-      -- will just be rendering markdown, so we handle them both the same way.
-      -- TODO these can have escaped/sanitized html codes in markdown. We
-      -- should make sure we handle this correctly.
+      -- The kind can be either plaintext or markdown.
+      -- If it's plaintext, then wrap it in a <text></text> block
 
       -- Some servers send input.value as empty, so let's ignore this :(
+      input.value = input.value or ''
+
+      if input.kind == "plaintext" then
+        -- very unlikely, but but escape the end tag, just to be sure
+        input.value = input.value:gsub("</text>", "\\</text>")
```

Comment:
Do we need this workaround? I feel like we should instead report this upstream in the language server than having a workaround for something which may/may not occur.

Reply:
You mean the escaping?

I can leave it out. Very unlikely it will ever occur. Happy to change it if you want.

Reply:
Not at home right now, but I'll remove it. Not really needed.

Reply:
kk will merge after

Reply:
done